core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1102,6 +1121,22 @@ public Builder setDefaultPartitionSpec(int specId) {
       return this;
     }
 
+    private Builder removePartitionSpec(PartitionSpec spec) {
+      Preconditions.checkArgument(
```

Comment:
This only catches us if we call removePartitionSpec after other operations so i'm not sure we should have it here? I'm just a little worried about the false sense of security.

Reply:
Sure sounds good to me